src/core/lib/event_engine/posix_engine/traced_buffer_list.cc

```diff
@@ -198,23 +199,27 @@ void ExtractOptStatsFromCmsg(ConnectionMetrics* metrics,
 }  // namespace.
 
 void TracedBufferList::AddNewEntry(int32_t seq_no, int fd, void* arg) {
-  buffer_list_.emplace_back(seq_no, arg);
-  TracedBuffer& new_elem = buffer_list_.back();
+  TracedBuffer new_elem(seq_no, arg);
   // Store the current time as the sendmsg time.
   new_elem.ts_.sendmsg_time.time = gpr_now(GPR_CLOCK_REALTIME);
   new_elem.ts_.scheduled_time.time = gpr_inf_past(GPR_CLOCK_REALTIME);
   new_elem.ts_.sent_time.time = gpr_inf_past(GPR_CLOCK_REALTIME);
   new_elem.ts_.acked_time.time = gpr_inf_past(GPR_CLOCK_REALTIME);
-
   if (GetSocketTcpInfo(&new_elem.ts_.info, fd) == 0) {
     ExtractOptStatsFromTcpInfo(&new_elem.ts_.sendmsg_time.metrics,
                                &new_elem.ts_.info);
   }
+  grpc_core::MutexLock lock(&mu_);
+  buffer_list_.push_back(new_elem);
```

Comment:
`new_elem.ts_.info` is a a very large structure and is not trivially copy-able.  Can we make return `next_` to `TraceBuffer`, allocate it on the heap and make TracedBufferList use an intrusive list approach to avoid this copy?

Reply:
fixed.